packages/cli/src/api/credentials.api.ts

```diff
@@ -0,0 +1,378 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable import/no-cycle */
+import express = require('express');
+import { getConnection, In } from 'typeorm';
+import { UserSettings, Credentials } from 'n8n-core';
+import { INodeCredentialTestResult } from 'n8n-workflow';
+
+import {
+	CredentialsHelper,
+	Db,
+	GenericHelpers,
+	ICredentialsDb,
+	ICredentialsResponse,
+	whereClause,
+} from '..';
+import * as ResponseHelper from '../ResponseHelper';
+
+import { RESPONSE_ERROR_MESSAGES } from '../constants';
+import { CredentialsEntity } from '../databases/entities/CredentialsEntity';
+import { SharedCredentials } from '../databases/entities/SharedCredentials';
+import { validateEntity } from '../GenericHelpers';
+import type { CredentialRequest } from '../requests';
+import config = require('../../config');
+import { externalHooks } from '../Server';
+
+export const credentialsController = express.Router();
+
+/**
+ * GET /credentials
+ */
+credentialsController.get(
+	'/',
+	ResponseHelper.send(async (req: CredentialRequest.GetAll): Promise<ICredentialsResponse[]> => {
+		let credentials: ICredentialsDb[] = [];
+
+		const filter = req.query.filter ? (JSON.parse(req.query.filter) as Record<string, string>) : {};
+
+		if (req.user.globalRole.name === 'owner') {
+			credentials = await Db.collections.Credentials!.find({
+				select: ['id', 'name', 'type', 'nodesAccess', 'createdAt', 'updatedAt'],
+				where: filter,
+			});
+		} else {
+			const shared = await Db.collections.SharedCredentials!.find({
+				where: whereClause({
+					user: req.user,
+					entityType: 'credentials',
+				}),
+			});
+
+			if (!shared.length) return [];
+
+			credentials = await Db.collections.Credentials!.find({
+				select: ['id', 'name', 'type', 'nodesAccess', 'createdAt', 'updatedAt'],
+				where: {
+					id: In(shared.map(({ credentialId }) => credentialId)),
+					...filter,
+				},
+			});
+		}
+
+		return credentials.map(({ id, ...rest }) => ({ id: id.toString(), ...rest }));
```

Comment:
no clue what would be more performant but wdyt about this:
```suggestion
		return credentials.map((credential) => Object.assign(credential, { id: credential.id.toString() }));
```